components/base/wheeled/kinematics.go

```diff
@@ -51,13 +61,74 @@ func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
-	return []referenceframe.Input{}, errors.New("not implemented yet")
+	// TODO(rb): make a transformation from the component reference to the base frame
+	pose, _, err := kwb.slam.GetPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	pt := pose.Point()
+	theta := math.Mod(pose.Orientation().OrientationVectorRadians().Theta, 2*math.Pi) - math.Pi
+	return []referenceframe.Input{{Value: pt.X}, {Value: pt.Y}, {Value: theta}}, nil
 }
 
-func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
-	// TODO(RSDK-2311): complete the implementation
-	return errors.New("not implemented yet")
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, inputs []referenceframe.Input) (err error) {
+	// create a frame system to be used during the context of this function, used to locate the goal in the base frame
+	fs := referenceframe.NewEmptyFrameSystem("")
+	if err := fs.AddFrame(kwb.model, fs.World()); err != nil {
+		return err
+	}
+
+	// create a function for the error state, which is defined as [positional error, heading error]
+	errorState := func() (int, float64, error) {
+		currentInputs, err := kwb.CurrentInputs(ctx)
+		if err != nil {
+			return 0, 0, err
+		}
+
+		// create a goal pose in the world frame
+		desiredHeading := math.Atan2(currentInputs[1].Value-inputs[1].Value, currentInputs[0].Value-inputs[0].Value)
+		goal := referenceframe.NewPoseInFrame(
+			referenceframe.World,
+			spatialmath.NewPose(
+				r3.Vector{X: inputs[0].Value, Y: inputs[1].Value},
+				&spatialmath.OrientationVector{OZ: 1, Theta: desiredHeading},
+			),
+		)
+
+		// transform the goal pose such that it is in the base frame
+		tf, err := fs.Transform(map[string][]referenceframe.Input{kwb.name: currentInputs}, goal, kwb.name)
+		if err != nil {
+			return 0, 0, err
+		}
+		delta, ok := tf.(*referenceframe.PoseInFrame)
+		if !ok {
+			return 0, 0, errors.New("can't interpret transformable as a pose in frame")
+		}
+
+		// calculate the error state
+		headingErr := math.Mod(delta.Pose().Orientation().OrientationVectorDegrees().Theta, 360)
+		positionErr := int(1000 * delta.Pose().Point().Norm())
+		kwb.logger.Debugf("Position error: \t%d mm\tHeading error: \t%f degrees", positionErr, headingErr)
+		return positionErr, headingErr, nil
+	}
+
+	// this loop polls the error state and issues a corresponding command to move the base to the objective
+	// when the base is within the positional threshold of the goal, exit the loop
+	// TODO(rb): check for the context being cancelled and stop if so
+	for distErr, headingErr, err := errorState(); err == nil && distErr > positionThresholdMM; distErr, headingErr, err = errorState() {
+		if math.Abs(headingErr) > headingThresholdDegrees {
+			// base is headed off course; spin to correct
+			err = kwb.Spin(ctx, -headingErr, 60, nil)
+		} else {
+			// base is pointed the correct direction; forge onward
+			err = kwb.MoveStraight(ctx, distErr, 300, nil)
```

Comment:
Is traveling this distance the same as traveling for 1 second? From the tech spec:
b. Poll the SLAM service at a specified frequency (1 Hz) and cancel the movement, overwriting with a new command

Reply:
I had originally tried to do it in this way but was having trouble getting this to work.  Now that I have a starting place I am much more able to iterate on it to get this behavior.  My inclination is to make this a separate PR, and if you're ok with that I will add it to the list in the description above as future work

Reply:
That sounds good! I also don't feel strongly whether we check position every 1 second or after each MoveStraight call, so I think we could leave as is and just update the tech spec. However, if we do that, I think we should pull the 300 out into a constant.

Reply:
In the interest of getting this merged ASAP, I made a callout in the description of the PR that this is future work.  Before I merge I will ticket all these remaining TODOs.  

I will remove the magic numbers in place of constants though, good idea

Reply:
sounds good